src/modules/mavlink/mavlink_mission.cpp

```diff
@@ -1261,16 +1271,16 @@ MavlinkMissionManager::parse_mavlink_mission_item(const mavlink_mission_item_t *
 {
 	if (mavlink_mission_item->frame == MAV_FRAME_GLOBAL ||
 	    mavlink_mission_item->frame == MAV_FRAME_GLOBAL_RELATIVE_ALT ||
-	    (_int_mode && (mavlink_mission_item->frame == MAV_FRAME_GLOBAL_INT ||
-			   mavlink_mission_item->frame == MAV_FRAME_GLOBAL_RELATIVE_ALT_INT))) {
+	    (int_mode() && (mavlink_mission_item->frame == MAV_FRAME_GLOBAL_INT ||
+			    mavlink_mission_item->frame == MAV_FRAME_GLOBAL_RELATIVE_ALT_INT))) {
 
 		// Switch to int mode if that is what we are receiving
-		if ((mavlink_mission_item->frame == MAV_FRAME_GLOBAL_INT ||
-		     mavlink_mission_item->frame == MAV_FRAME_GLOBAL_RELATIVE_ALT_INT)) {
-			_int_mode = true;
-		}
+//		if ((mavlink_mission_item->frame == MAV_FRAME_GLOBAL_INT ||
+//		     mavlink_mission_item->frame == MAV_FRAME_GLOBAL_RELATIVE_ALT_INT)) {
+//			int_mode() = true;
```

Comment:
Here comes the catch: we somehow still need to be compatible to older ground stations which do not support micro services yet. So, we need some sort of "if no microservice version chosen, then try to guess if int or not".